### PR TITLE
Phi2 changes

### PR DESCRIPTION
**Pull request by wangyems**
## Describe your changes

prerequisite PR: https://github.com/microsoft/onnxruntime/pull/19500

Phi2 changes:
1. Fixed dynamo export bugs
2. Leveraged most updated optimization script.
3. Provided cpu_fp32, cpu_int4, cuda_fp16 and cuda_int4 passes
4. Readme modifications.

### CODE REVIEW

File: examples/phi2/phi2_optimize_template.json
Comment (jambayk) [inline examples/phi2/phi2_optimize_template.json:128]: could you add the desired ep like in the llama2 example https://github.com/microsoft/Olive/blob/main/examples/llama2/llama2.py#L106 

Given an EP, Olive can infer the corresponding accelerator. So, I think for the sake of simplicity, we can remove the `target` and `systems` part entirely. The system is local, so it is redundant. Thanks!

File: examples/phi2/user_script.py
Comment (jambayk) [inline examples/phi2/user_script.py:219]: Does the dynamo exporter change the name of the inputs? Or it is a result of the transformers optimization pass?

File: olive/passes/onnx/conversion.py
Comment (jambayk) [inline olive/passes/onnx/conversion.py:183]: Is there a reason to move this outside the `if` block? It doesn't look like dynamo exporter requires any of the io_config, other than the dummy imput cleanup. Which in this case could be fixed using the user script if required.

File: examples/phi2/user_script.py
Comment (wangyems) [reply examples/phi2/user_script.py:219]: it's the result of the transformers optimization pass

File: olive/passes/onnx/conversion.py
Comment (wangyems) [reply olive/passes/onnx/conversion.py:183]: I think that the shared implementation of dummy input cleanup keeps the user_script be maintained regardless of export approach. Since output_names and dynamic_axes are used in script export only, let me move them inside the if block.

File: examples/phi2/phi2_optimize_template.json
Comment (wangyems) [reply examples/phi2/phi2_optimize_template.json:128]: fixed

**Comment by wangyems** [comment]
@jambayk Thanks for the review! I've updated it based on your comments.

File: olive/passes/onnx/conversion.py
Comment (jambayk) [inline olive/passes/onnx/conversion.py:226]: Thanks for the changes! Just one more comment:

Since the unused dummy input delete steps happen in both cases, can you move it to line 212? Like before https://github.com/microsoft/Olive/blob/main/olive/passes/onnx/conversion.py#L234?

**Comment by jambayk** [comment]
We have some linters set up for the repo. The instructions to run them locally can be found here https://github.com/microsoft/Olive/blob/main/CONTRIBUTING.md#linting

File: olive/passes/onnx/conversion.py
Comment (wangyems) [reply olive/passes/onnx/conversion.py:226]: there is a slight difference in deletion. for input type, script export accepts map but dynamo export accepts tuple
dynamo:
```
for key in unused_keys:
     dummy_inputs[key] = None
```
script:
```
for key in unused_keys:
    del dummy_inputs[key]
```

Comment (jambayk) [reply olive/passes/onnx/conversion.py:226]: Oh I see, thanks!

File: olive/passes/onnx/transformer_optimization.py
Comment (jambayk) [inline olive/passes/onnx/transformer_optimization.py:130]: This config param is not used since the actual implementation gets it from `optimization_options` which itself if a config param.

File: olive/passes/onnx/transformer_optimization.py
Comment (wangyems) [reply olive/passes/onnx/transformer_optimization.py:189]: fixed

File: olive/passes/onnx/transformer_optimization.py
Comment (wangyems) [reply olive/passes/onnx/transformer_optimization.py]: removed

File: examples/phi2/readme.md
Comment (jambayk) [inline examples/phi2/readme.md:8]: Instead of linking the readme from the ort example, could you copy over the instructions for installation instructions for torch and ort-nightly? 
The linked prerequisite section has some more instructions about running the model conversion which might be confusing here. 

Other than that, the rest of the PR looks good to me. Just waiting for the related ORT PR to merge and the nightly package to be published before we merge this one. Thanks!

Comment (wangyems) [reply examples/phi2/readme.md:8]: Great! I'll let you know once the ORT PR is in

**Comment by wangyems** [comment]
The ORT change is merged. I'll check the nightly package tomorrow(2/15) and if all good we can go ahead and merge this PR.

**Review by jambayk — APPROVED** [review]

**Pull request merged**
